packages/mdc-checkbox/_variables.import.scss

```diff
@@ -1,7 +1,5 @@
-@forward "@material/theme/constants" as mdc-theme-*;
-@forward "@material/theme/variables" as mdc-theme-*;
-@forward "@material/density/variables" as mdc-density-*;
-@forward "@material/theme/functions" as mdc-theme-*;
-@forward "variables" as mdc-checkbox-*;
+@forward "@material/ripple" as mdc-* hide $mdc-dark-ink-opacities, $mdc-fade-in-duration, $mdc-fade-out-duration, $mdc-light-ink-opacities, $mdc-pressed-dark-ink-opacity, $mdc-pressed-light-ink-opacity, $mdc-translate-duration, mdc-core-styles, mdc-common, mdc-surface, mdc-states-base-color, mdc-states-opacities, mdc-states-hover-opacity, mdc-states-focus-opacity, mdc-states-focus-opacity-properties-, mdc-states-press-opacity, mdc-states, mdc-states-activated, mdc-states-selected, mdc-radius-bounded, mdc-radius-unbounded, mdc-states-interactions-, mdc-target-selector, mdc-target-common, mdc-keyframes-, mdc-states-opacity, mdc-states-opacities-;
+@forward "@material/ripple" as mdc-ripple-* hide $mdc-ripple-states-wash-duration, mdc-ripple-core-styles, mdc-ripple-common, mdc-ripple-surface, mdc-ripple-states-base-color, mdc-ripple-states-opacities, mdc-ripple-states-hover-opacity, mdc-ripple-states-focus-opacity, mdc-ripple-states-focus-opacity-properties-, mdc-ripple-states-press-opacity, mdc-ripple-states, mdc-ripple-states-activated, mdc-ripple-states-selected, mdc-ripple-radius-bounded, mdc-ripple-radius-unbounded, mdc-ripple-states-interactions-, mdc-ripple-target-selector, mdc-ripple-target-common, mdc-ripple-keyframes-, mdc-ripple-states-opacity, mdc-ripple-states-opacities-;
+@forward "@material/theme" as mdc-theme-*;
+@forward "@material/density" as mdc-density-* hide mdc-density-prop-value;
+@forward "./index" as mdc-checkbox-* hide $mdc-checkbox-ripple-target, mdc-checkbox-core-styles, mdc-checkbox-without-ripple, mdc-checkbox-ripple, mdc-checkbox-density, mdc-checkbox-ripple-size, mdc-checkbox-container-colors, mdc-checkbox-disabled-container-colors, mdc-checkbox-ink-color, mdc-checkbox-disabled-ink-color, mdc-checkbox-focus-indicator-color, mdc-checkbox-touch-target, mdc-checkbox-base-, mdc-checkbox-disabled-, mdc-checkbox-child--upgraded-, mdc-checkbox-anim-, mdc-checkbox-if-enabled-, mdc-checkbox-if-disabled-, mdc-checkbox-if-unmarked-enabled-, mdc-checkbox-if-unmarked-disabled-, mdc-checkbox-if-marked-enabled-, mdc-checkbox-if-marked-disabled-, mdc-checkbox-background-, mdc-checkbox-background--marked-, mdc-checkbox-container-colors-, mdc-checkbox-focus-indicator-, mdc-checkbox-focus-indicator--focused-, mdc-checkbox-native-control-, mdc-checkbox-checkmark-, mdc-checkbox-checkmark--checked-, mdc-checkbox-checkmark--indeterminate-, mdc-checkbox-ink-color-, mdc-checkbox-checkmark-path-, mdc-checkbox-checkmark-path--marked-, mdc-checkbox-mixedmark-, mdc-checkbox-mixedmark--checked-, mdc-checkbox-mixedmark--indeterminate-, mdc-checkbox-touch-target-reset-, mdc-checkbox-container-keyframes-, mdc-checkbox-mark-keyframes-, mdc-checkbox-transition-enter, mdc-checkbox-transition-exit;
```

Comment:
Why is it hiding some of the mixins starting with mdc-checkbox- prefix? Example: `mdc-checkbox-ripple` etc.

Same for other components.

Reply:
The goal is to forward exactly the same set of names that were forwarded previously. In the case of this file, only the `variables` file was forwarded previously, so all the mixins and functions (as well as `$mdc-checkbox-ripple-target` which is defined in `_mixins.scss`) are hidden to preserve the existing behavior.

Reply:
Can we not use index paths in import files in that case? I do not see any benefit of using index files here. This import path is very specific to variables might be a good idea to scope it to variables only. WDYT?

Can migrator script handle this without making a big change?

Reply:
When the migrator is generating new `@use` rules for stylesheets that depend on MDC Web, it figures out which URLs to use based on the existing `@forward` rules in the `.import.scss` files. Forwarding index files here means that downstream stylesheets migrated using the migrator will automatically use index files—otherwise they'd default to using the same old mixins, variables, and functions files.

We could potentially special-case this for the migration within Google in particular, but in the external world these `@forward` rules are the only way to communicate the correct pattern to downstream invocations of the migrator.

Reply:
I do not have any strong preference on how import file `@forward`s things from.

IMO, Since we intend to only forward variables here, using the direct path (Eg: `@material/checkbox/variables`) seems more idiomatic and avoids hiding lot of things.

IIUC, downstream users will not notice whether it is `@forward`-ed from index file or variables file.

Reply:
> IIUC, downstream users will not notice whether it is `@forward`-ed from index file or variables file.

They will notice when they run the migrator. If this file forwards from a variable file, the migrator will generate `@use "@material/checkbox/variables"`, but if this file forwards from an index file, the migrator will generate `@use "material/checkbox"`.

Reply:
Thanks for clarifying! If the migrator dependents on this import path then we can use index paths like you did.

Resolving this comment.